torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2282,9 +2282,21 @@ def track_symint(source, val, constraint=None):
                 elif isinstance(-s, sympy.Symbol):
                     symbol_to_source[-s].append(NegateSource(source))
                 else:
-                    if constraint is not None:
-                        # TODO: Maybe non-strict constraint shouldn't error
-                        # here?  Check what happens in practice
+                    constraint_violated = False
+                    if isinstance(constraint, StrictMinMaxConstraint):
+                        constraint_violated = True
+                    elif isinstance(constraint, RelaxedUnspecConstraint):
```

Comment:
Looks pretty similar, can you cook out a helper function?

Reply:
it's not exactly the same though. i'll look in a follow up.